exporter/collector/config.go

```diff
@@ -96,7 +96,10 @@ type MetricConfig struct {
 	// exporter. It allows overriding the function used to map otel resource to
 	// monitored resource.
 	MapMonitoredResource func(pcommon.Resource) *monitoredrespb.MonitoredResource
-	Prefix               string `mapstructure:"prefix"`
+	// ModifyMetrics is an extension point for exporters to directly modify the set
+	// of ResourceMetrics during the call to PushMetrics.
```

Comment:
Note that this means we need to mark the exporter as modifying the metrics (means an extra copy before export).  This may be OK for now.  Are there other ways we might be able to accomplish this?

Reply:
See https://github.com/open-telemetry/opentelemetry-collector/blob/main/consumer/consumer.go#L22

Reply:
Is there any way around this if we want to inject a new metric? eg, only making this insert the metric rather than an arbitrary modification path. Or is even that going to require this update?

Reply:
You might be able to Change from a `func ModifyMetrics(pmetric.Metrics)` to `func ExtraMetrics(pmetric.Metrics) pmetric.MetricSlice`, where the metricslice isn't ever actually added to the resource metrics, but is used _as if_ it was part of the resource metrics

Reply:
I ended up going with a copy approach, I think that leaves the original metrics in tact. ptal if that's right